### PR TITLE
BAEL-5192 Code samples for the article

### PR DESCRIPTION
**Pull request by yavuztas**

### CODE REVIEW

File: jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Test
+    public void givenInteger_deserializedAsBooleanByDefault() throws Exception {
+        // Integer "1" and "0" values deserialized correctly out of the box.
+        // No configuration or @JsonFormat annotation needed.
+        String json = "{\"id\":1,\"name\":\"My Game\",\"paused\":1,\"over\":0}";
+        Game game = mapper.readValue(json, Game.class);
+
+        assertThat(game.isPaused()).isEqualTo(true);
```
Comment (KevinGilmore) [inline jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java:42]: You could simply use the standard assertTrue here and assertFalse on the line below.

File: jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    }
+
+    @Test
+    public void givenNumericStringWithCustomDeserializer_deserializedAsBoolean() throws Exception {
+        String json = "{\"id\":1,\"name\":\"My Game\",\"paused\":\"1\",\"over\":\"0\"}";
+        GameAnnotatedByJsonSerializeDeserialize
+          game = mapper.readValue(json, GameAnnotatedByJsonSerializeDeserialize.class);
+
+        assertThat(game.isPaused()).isEqualTo(true);
+        assertThat(game.isOver()).isEqualTo(false);
```
Comment (KevinGilmore) [inline jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java:80]: Use assertTrue and assertFalse

File: jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+        // setting deserializers globally
+        SimpleModule module = new SimpleModule();
+        module.addDeserializer(Boolean.class, new NumericBooleanDeserializer());
+        mapper.registerModule(module);
+
+        String json = "{\"id\":1,\"name\":\"My Game\",\"paused\":\"1\",\"over\":\"0\"}";
+        Game game = mapper.readValue(json, Game.class);
+
+        assertThat(game.isPaused()).isEqualTo(true);
+        assertThat(game.isOver()).isEqualTo(false);
```
Comment (KevinGilmore) [inline jackson-modules/jackson-conversions-2/src/test/java/com/baeldung/jackson/booleanAsInt/BooleanAsIntegerUnitTest.java:110]: Use assertTrue and assertFalse

**Review by KevinGilmore — DISMISSED** [review]
Please disregard my notes here.

**Comment by lor6** [comment]
retest this please

**Pull request merged**